models/models.go

```diff
@@ -74,44 +73,44 @@ type Advisory struct {
 // Cve : >definitions>definition>metadata>advisory>cve
 // RedHat OVAL
 type Cve struct {
-	ID         uint `gorm:"primary_key"`
-	AdvisoryID uint `json:"-" xml:"-"`
-
-	CveID  string
-	Cvss2  string
-	Cvss3  string
-	Cwe    string
-	Impact string
-	Href   string
-	Public string
+	ID         uint `gorm:"primary_key" json:"-"`
+	AdvisoryID uint `gorm:"idx_cves_advisory_id" json:"-" xml:"-"`
+
+	CveID  string `gorm:"type:varchar(255)"`
+	Cvss2  string `gorm:"type:varchar(255)"`
+	Cvss3  string `gorm:"type:varchar(255)"`
+	Cwe    string `gorm:"type:varchar(255)"`
+	Impact string `gorm:"type:varchar(255)"`
+	Href   string `gorm:"type:varchar(255)"`
+	Public string `gorm:"type:varchar(255)"`
 }
 
 // Bugzilla : >definitions>definition>metadata>advisory>bugzilla
 // RedHat OVAL
 type Bugzilla struct {
-	ID         uint `gorm:"primary_key"`
-	AdvisoryID uint `json:"-" xml:"-"`
+	ID         uint `gorm:"primary_key" json:"-"`
+	AdvisoryID uint `gorm:"index:idx_bugzillas_advisory_id" json:"-" xml:"-"`
 
-	BugzillaID string
-	URL        string
-	Title      string
+	BugzillaID string `gorm:"type:varchar(255)"`
+	URL        string `gorm:"type:varchar(255)"`
+	Title      string `gorm:"type:varchar(255)"`
 }
 
 // Cpe : >definitions>definition>metadata>advisory>affected_cpe_list
 type Cpe struct {
-	ID         uint `gorm:"primary_key"`
-	AdvisoryID uint `json:"-" xml:"-"`
+	ID         uint `gorm:"primary_key" json:"-"`
+	AdvisoryID uint `gorm:"index:idx_cpes_advisory_id" json:"-" xml:"-"`
 
-	Cpe string
+	Cpe string `gorm:"type:varchar(255)"`
 }
 
 // Debian : >definitions>definition>metadata>debian
 type Debian struct {
-	ID           uint `gorm:"primary_key"`
-	DefinitionID uint `json:"-" xml:"-"`
+	ID           uint `gorm:"primary_key" json:"-"`
+	DefinitionID uint `gorm:"index:idx_debian_definition_id" json:"-" xml:"-"`
 
-	CveID    string
-	MoreInfo string `sql:"type:text"` // https://github.com/jinzhu/gorm/issues/510#issuecomment-180669092
+	CveID    string `gorm:"type:varchar(255);index:idx_debian_cve_id"`
+	MoreInfo string `gorm:"type:text"`
```

Comment:
Is this text struct tag required?

Reply:
Don't you want all columns to be allocated with longtext?